src/SecurityChecks.js

```diff
@@ -0,0 +1,150 @@
+        });
+      }
+    }
+    clpWarnings[className] = thisClassWarnings;
+  }
+  return clpWarnings;
+}
+function ServerConfig(req) {
+  const options = req.config || req;
+  const warnings = [];
+  if (options.allowClientClassCreation) {
+    warnings.push({
+      title: 'Allow Client Class Creation is not recommended.',
+      message:
+        'Allow client class creation is not recommended for production servers it allows any user - authorized or not - to create a new class.',
+      link: 'https://docs.parseplatform.org/js/guide/#restricting-class-creation',
+    });
+  }
+  if (!options.masterKey.match('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{14,})')) {
+    warnings.push({
+      title: 'Weak masterKey.',
+      message:
+        'masterKey is a key that overrides all permissions. You should use a secure string for your masterKey',
+      link: 'https://docs.parseplatform.org/parse-server/guide/#keys',
+    });
+  }
+  let https = false;
+  try {
+    const serverURL = url.parse(options.serverURL);
+    https = serverURL.protocol === 'https:';
+  } catch (e) {
+    /* */
+  }
+  if (!https) {
+    warnings.push({
+      title: `Server served over HTTP`,
+      message: 'We strongly recommend using a HTTPS protocol.',
+    });
+  }
+  return warnings;
+}
+async function Files(req) {
+  const options = req.config || req;
+  const fileTrigger = getTrigger('@File', 'beforeSaveFile', options.appId);
+  const fileWarnings = [];
+  if (!fileTrigger) {
+    fileWarnings.push({
+      title: `No beforeFileSave Trigger`,
+      message:
+        "Even if you don't store files, we strongly recommend using a beforeFileSave trigger to prevent unauthorized uploads.",
+      link: 'https://docs.parseplatform.org/cloudcode/guide/#beforesavefile',
+    });
+  } else {
+    try {
+      const file = new Parse.File('testpopeye.txt', [1, 2, 3], 'text/plain');
+      await file.save();
+      fileWarnings.push({
+        title: `Unrestricted access to file uploads`,
+        message:
+          'Even though you have a beforeFileSave trigger, it allows unregistered users to upload.',
+        link: 'https://docs.parseplatform.org/cloudcode/guide/#beforesavefile',
+      });
+      await options.filesController.deleteFile(file._name);
+    } catch (e) {
+      /* */
+    }
+  }
+  return fileWarnings;
+}
+
+export async function securityChecks(req) {
+  try {
+    const options = req.config || req;
+    if (!options.securityChecks.enabled) {
+      return { error: { code: 1, error: 'Security checks are not enabled.' } };
+    }
+    const functions = {
+      CLP,
+      ServerConfig,
+      Files,
+    };
+    if (
+      options.databaseAdapter.getSecurityLogs &&
+      typeof options.databaseAdapter.getSecurityLogs === 'function'
+    ) {
+      functions.Database = options.databaseAdapter.getSecurityLogs;
+    }
+    const response = {};
+    let totalWarnings = 0;
+    for (const name in functions) {
```

Comment:
Can we simplify this by removing the loop and array of function names? I also think, if you always returned an array from a check function, you wouldn't have to test whether it's an array.

Reply:
Sure, I was thinking that for in the future, for a developer to add a new test, they would just have to add it to the functions object, and the rest of the code handles the counts etc.

CLP security recommendations are an Object I believe.

Reply:
I think there is still room for simplification, for example:

```
const functions = {
    CLP: checkCLP,
    ServerConfig: checkServerConfig,
    Files: checkFiles,
};
```
and
```
if (Array.isArray(result)) {
    totalWarnings += result.length;
} else {
    totalWarnings += Object.keys(result).length;
}
```